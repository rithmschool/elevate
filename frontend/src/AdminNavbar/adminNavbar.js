import React from "react";

class AdminNavbar extends React.Component {
  handleClick = evt => {
    evt.preventDefault();
    this.props.changeView(evt.target.id);
  }

  render(){
    return (
      <div className="adminNavbar">
        <div id="users" onClick={this.handleClick}>Users</div>
        <div id="questions" onClick={this.handleClick}>Questions</div>
        <div id="invoices" onClick={this.handleClick}>Invoices</div>
        <div id="calendar" onClick={this.handleClick}>Calendar</div>
      </div>
    )
  }
}



export default AdminNavbar;
