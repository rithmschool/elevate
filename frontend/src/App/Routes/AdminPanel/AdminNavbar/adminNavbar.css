.adminNavbar > p {
  cursor: pointer;
  padding: 5px;
}

.adminNavbar > p > #questions,
#users,
#invoices,
#calendar {
  text-decoration: none;
  color: lightgrey;
}

.adminNavbar {
  height: 100vh;
  margin: 18px 6px;
}
