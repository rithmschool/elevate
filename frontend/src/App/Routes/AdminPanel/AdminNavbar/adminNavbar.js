import React from "react";
import { Link } from "react-router-dom";
import "./adminNavbar.css";

class AdminNavbar extends React.Component {
  toggle = () => {
    this.props.toggleSidebar();
  };

  render() {
    let view = this.props.position ? (
      <div className="adminNavbar">
        <p>
          <Link id="users" className="admin-link" to="/admin/users">
            Users
          </Link>
        </p>
        <p>
          <Link id="questions" className="admin-link" to="/admin/questions">
            Questions
          </Link>
        </p>
        <p>
          <Link id="invoices" className="admin-link" to="/admin/invoices">
            Invoices
          </Link>
        </p>
        <p>
          <Link id="calendar" className="admin-link" to="/admin/calendar">
            Calendar
          </Link>
        </p>
      </div>
    ) : (
      <div onClick={this.toggle} className="adminNavbar"></div>
    );

    return view;
  }
}

export default AdminNavbar;
