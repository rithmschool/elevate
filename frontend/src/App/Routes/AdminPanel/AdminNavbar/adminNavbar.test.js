import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AdminNavbar from "./adminNavbar";

describe("AdminNavbar", function() {
  let wrapper;

  beforeEach(function() {
    wrapper = shallow(<AdminNavbar position={true} />);
  });

  it("renders without crashing", function() {
    shallow(<AdminNavbar />);
  });

  it("matches snapshot", function() {
    let serialized = toJson(wrapper);

    expect(serialized).toMatchSnapshot();
  });

  it("has div with admin-navbar class", function() {
    expect(wrapper.find("div.adminNavbar")).toHaveLength(1);
  });

  it("should be able to call toggle method when position is false", function() {
    const spy = jest.spyOn(wrapper.instance(), "toggle");

    wrapper.setProps({
      toggleSidebar: function toggleSidebar() {},
      position: false
    });

    wrapper.find(".adminNavbar").simulate("click");

    expect(spy).toHaveBeenCalled();
  });

  it("should display 'users' and 'questions' when position is true", function() {
    // wrapper.setProps({ position: true });

    expect(wrapper.find(".adminNavbar > p > #users")).toHaveLength(1);
    expect(wrapper.find(".adminNavbar > p > #questions")).toHaveLength(1);
  });
});
