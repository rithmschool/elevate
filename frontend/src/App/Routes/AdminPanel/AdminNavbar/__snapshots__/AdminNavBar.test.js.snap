// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminNavbar matches snapshot 1`] = `
<AdminNavbar>
  <div
    className="adminNavbar"
    onClick={[Function]}
  />
</AdminNavbar>
`;
